repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -333,7 +340,7 @@ def prepare_target_userspace(context, userspace_dir, enabled_repos, packages):
         # transaction check phase is done - so the preupgrade checks won't affect the host system.
         # The 'switch_cln_channel_download' actor should take care of switching the channel back to the CL8 channel
         # when it's time to download the upgrade packages.
-        cln_switch(target=8)
+        cln_switch(target=int(get_target_major_version()))
```

Comment:
`target_major_version` defined [above](https://github.com/cloudlinux/leapp-repository/pull/30/files#diff-2ff1208b2da0c18d7cc064b75ade308aa8a16e7eb1779a0ec7a602f76b692b6cR255) can be reused here.